eventuals/task.h

```diff
@@ -570,57 +598,66 @@ class TaskFromToWith {
 
 ////////////////////////////////////////////////////////////////////////
 
-struct Task final {
-  template <typename From_>
-  struct From final : public TaskFromToWith<From_, void> {
-    template <typename To_>
-    struct To final : public TaskFromToWith<From_, To_> {
-      template <typename... Args_>
-      using With = TaskFromToWith<From_, To_, Args_...>;
-
-      To(MonostateIfVoidOr<To_> value)
-        : TaskFromToWith<From_, To_>(std::move(value)) {}
-
-      template <typename F>
-      To(F f)
-        : TaskFromToWith<From_, To_>(std::move(f)) {}
+template <
+    typename From_ = void,
+    typename To_ = void,
+    typename Errors_ = std::tuple<>,
+    typename... Args_>
+struct TaskBuilder final
+  : public TaskFromToWith<From_, To_, Errors_, Args_...> {
+  template <typename T>
+  using From = TaskBuilder<T, To_, Errors_, Args_...>;
 
-      To(To&& that) = default;
+  template <typename T>
+  using To = TaskBuilder<From_, T, Errors_, Args_...>;
 
-      ~To() override = default;
-    };
+  template <typename T>
+  using Of = TaskBuilder<void, T, Errors_, Args_...>;
 
-    template <typename... Args_>
-    using With = TaskFromToWith<From_, void, Args_...>;
+  template <typename... Args>
+  using With = TaskBuilder<From_, To_, Errors_, Args...>;
 
-    template <typename F>
-    From(F f)
-      : TaskFromToWith<From_, void>(std::move(f)) {}
+  template <typename... Errors>
+  using Raises = TaskBuilder<From_, To_, std::tuple<Errors...>, Args_...>;
 
-    From(From&& that) = default;
+  template <typename F>
+  TaskBuilder(Args_... args, F f)
+    : TaskFromToWith<
+        From_,
+        To_,
+        Errors_,
+        Args_...>(std::move(args)..., std::move(f)) {}
 
-    ~From() override = default;
-  };
+  ~TaskBuilder() = default;
 
-  template <typename To_>
-  using Of = From<void>::To<To_>;
+  TaskBuilder(TaskBuilder&&) = default;
 
-  template <typename... Args_>
-  using With = From<void>::To<void>::With<Args_...>;
+  TaskBuilder(MonostateIfVoidOr<To_> value)
+    : TaskFromToWith<From_, To_, Errors_, Args_...>(std::move(value)) {}
 
   // Helpers for synchronous tasks.
-  template <typename Value>
+  template <typename Value, typename... Errors>
   static auto Success(Value value) {
-    return TaskFromToWith<void, Value>(std::move(value));
+    return TaskFromToWith<
+        void,
+        Value,
+        std::tuple<Errors...>>(std::move(value));
   }
 
-  template <typename Value>
+  template <typename Value, typename... Errors>
   static auto Success(std::reference_wrapper<Value> value) {
-    return TaskFromToWith<void, Value&>(std::move(value));
+    return TaskFromToWith<
+        void,
+        Value&,
+        std::tuple<Errors...>>(std::move(value));
```

Comment:
Errors is always empty! See [here in expected.h](https://github.com/3rdparty/eventuals/blob/main/eventuals/expected.h#L193).